Add pre-commit hooks to check and fix deprecations

jairus-m opened the pull request:
## Summary
Addresses #171. 

Based off the implementation patterns found in the classic [pre-commit](https://github.com/pre-commit/pre-commit-hooks) repo, this PR adds two pre-commit hooks to allow for automatic  updates of deprecated configs:
- dbt-autofix-check
- dbt-autofix-fix

Therefore, all someone would need to add in their local `.pre-commit-config.yaml` would be:

```yaml
repos:
  - repo: https://github.com/dbt-labs/dbt-autofix
    rev: v0.13.x # or 'main' or 'HEAD'
    hooks:
      - id: dbt-autofix-check # or dbt-autofix-fix
```
## Tests
I was able to test my fork with these changes... and it worked! 

<img width="1013" height="681" alt="Screenshot 2025-10-19 at 2 55 44 PM" src="https://github.com/user-attachments/assets/c9c547c6-c43e-471e-b913-b965d38c1a46" />

## Questions
1. Does `dbt-autofix` want to be used in this way?
2. Should there be a unit test(s) for the existing logic in `pre_commit_hooks/check_deprecations.py` ?

b-per commented on pyproject.toml:
```diff
@@ -33,7 +33,7 @@ test = [
 ]
 
 [tool.pdm.build]
-includes = ["dbt_autofix"]
```
I'm not familiar with packaging hooks in repos.
Would you have any docs mentioning why we need to add `pre_commit_hooks`?

jairus-m commented:
Thanks for taking a look @b-per ! I made some key updates today and addressed all the comments thus far. I re-pointed a real project to my forked branch and was able to test the following on some mock deprecations that i staged and dry ran with `pre-commit run`:

1. dbt-autofix-check hook
2. dbt-autofix-fix hook
3. Passing flags
    - --include-packages flag
    - --behavior-change flag
4. Passing multiple flags combined

Given that the implementation of [handling custom dbt directory paths](https://github.com/dbt-labs/dbt-autofix/pull/172/commits/fe7546fea3a347e0ccc323c4191188c251000ee3) re-uses existing logic in `dbt-autofix`, I do think it will work but candidly have not tested that yet.
